ice-phishing-detection-bot/src/agent.js

```diff
@@ -0,0 +1,142 @@
+const { Finding, FindingSeverity, FindingType } = require("forta-agent");
+const {
+  DEX_AND_CEX_ADDRESSES,
+  ApprovalThreshold,
+  ApprovalTimePeriod,
+} = require("./agent.config");
+const AddressApprovalTracker = require("./AddressApprovalTracker");
+const eventABIs = [
+  "event Approval(address indexed owner,address indexed spender,uint256 value)",
+  "event Transfer(address indexed from, address indexed to, uint256 value)",
+];
+
+const addressesTracked = [];
+let isRunning = false;
+let result = [];
+const provideHandleTransaction = (addressesTracked) => {
+  return async function handleTransaction(txEvent) {
+    const findings = [];
+
+    if (!DEX_AND_CEX_ADDRESSES.includes(txEvent.from)) {
+      const filtered = txEvent.filterLog(eventABIs);
+
+      for (let tx of filtered) {
+        const { name } = tx;
+        const to = txEvent.to;
+        const from = txEvent.from;
+        const hash = txEvent.hash;
+
+        const foundIndex = addressesTracked.findIndex(
```

Comment:
Do we need to store the tracked addresses in an array? Wouldn't it be easier and faster to store them in a map/object?